pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/retention/RetentionManager.java

```diff
@@ -43,53 +42,44 @@
  * The <code>RetentionManager</code> class manages retention for all segments and delete expired segments.
  * <p>It is scheduled to run only on leader controller.
  */
-public class RetentionManager {
+public class RetentionManager extends ControllerPeriodicTask {
   public static final long OLD_LLC_SEGMENTS_RETENTION_IN_MILLIS = TimeUnit.DAYS.toMillis(5L);
 
   private static final Logger LOGGER = LoggerFactory.getLogger(RetentionManager.class);
 
-  private final PinotHelixResourceManager _pinotHelixResourceManager;
-  private final ScheduledExecutorService _executorService;
-  private final int _runFrequencyInSeconds;
   private final int _deletedSegmentsRetentionInDays;
 
   public RetentionManager(PinotHelixResourceManager pinotHelixResourceManager, int runFrequencyInSeconds,
       int deletedSegmentsRetentionInDays) {
-    _pinotHelixResourceManager = pinotHelixResourceManager;
-    _runFrequencyInSeconds = runFrequencyInSeconds;
+    super("PinotRetentionManager", runFrequencyInSeconds, Math.min(60, runFrequencyInSeconds),
+        pinotHelixResourceManager);
     _deletedSegmentsRetentionInDays = deletedSegmentsRetentionInDays;
-    _executorService =
-        Executors.newSingleThreadScheduledExecutor(r -> new Thread(r, "PinotRetentionManagerExecutorService"));
   }
 
-  public void start() {
+  @Override
+  public void onBecomeLeader() {
     LOGGER.info("Starting RetentionManager with runFrequencyInSeconds: {}, deletedSegmentsRetentionInDays: {}",
-        _runFrequencyInSeconds, _deletedSegmentsRetentionInDays);
-    _executorService.scheduleWithFixedDelay(this::execute, Math.min(60, _runFrequencyInSeconds), _runFrequencyInSeconds,
-        TimeUnit.SECONDS);
-    LOGGER.info("RetentionManager started");
+        getIntervalInSeconds(), _deletedSegmentsRetentionInDays);
```

Comment:
We should not need run-frequency here, right? It can be logged in the base class for all the periodic tasks.

Reply:
The purpose of the log here is to log the value of `_deletedSegmentsRetentionInDays` rather than run-frequency. The latter one has been added to the base class.